site/app/controllers/admin/ReportController.php

```diff
@@ -67,6 +67,14 @@ public function generateGradeSummaries() {
             $this->core->redirect($this->core->buildCourseUrl(['reports']));
         }
 
+        $poll_base_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), 'reports');
+
+        // Check that the directory is writable, fail if not
+        if (!is_writable($poll_base_path)) {
```

Comment:
I was suggestion moving the entire if block here, and not just the error check, but this is fine. I do think we still want to run this error check only if polls are enabled or else in an incorrectly configured setup, someone might hit this error with polls disabled, and then complain about this as a bug that polls are doing something they shouldn't be.

```suggestion
        if ($this->core->getConfig()->isPollsEnabled() && !is_writable($poll_base_path)) {
```